tests/kernel/ebpf.pm

```diff
@@ -0,0 +1,101 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Executes eBPF kselftests
+# Maintainer: Kernel QE <kernel-qa@suse.de>
+
+use base 'opensusebasetest';
+
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use registration;
+use utils;
+use LTP::WhiteList;
+use LTP::kirk;
+use LTP::utils;
+
+sub run_tests
+{
+    my ($root) = @_;
+
+    my $env = LTP::utils::prepare_whitelist_environment();
+    my $issues = get_var('KSELFTESTS_KNOWN_ISSUES', '');
+    my $whitelist = LTP::WhiteList->new($issues);
+    my @skipped = $whitelist->list_skipped_tests($env, 'kselftests');
+    my $skip_tests;
+    if (@skipped) {
+        $skip_tests = join("|", @skipped);
+
+        record_info(
+            "Exclude",
+            "Excluding tests: $skip_tests",
+            result => 'softfail'
+        );
+    }
+
+    my @volumes = (
+        {src => $root, dst => $root},
+        {src => "/tmp", dst => "/tmp"}
+    );
+
+    LTP::kirk->run(
+        framework => "kselftests:root=$root",
+        skip => $skip_tests,
+        suite => "bpf",
+        # when KIRK_INSTALL == 'container' we want to share
+        # kselftests folder and kirk logs folder
+        container_volumes => \@volumes,
+    );
+}
+
+sub ebpf_from_source
+{
+    # add repo with newest version of clang
+    zypper_call("ar -f http://download.suse.de/ibs/SUSE:/SLE-15-SP6:/GA/standard/ GA_standard");
```

Comment:
Which packages are missing in SLE and we need to add whole repository?  Also hardcoded url just for one version is not good  idea.

Reply:
I might remove compiling tools from source completely, because from one kernel version to an another we might have to change everything or to apply kernel patches.